src/Views/BucketPanel/EntryDetail.tsx

```diff
@@ -114,6 +126,102 @@ export default function EntryDetail(props: Readonly<Props>) {
     }
   };
 
+  const handleEditLabels = (record: any) => {
+    setCurrentRecord(record);
+
+    // Format the labels exactly like the When Condition editor
+    let labelsJson = "{}";
+
+    if (record.labels) {
+      try {
+        // First convert to string if it's an object
+        const labelsStr =
+          typeof record.labels === "string"
+            ? record.labels
+            : JSON.stringify(record.labels);
+
+        // Parse and reformat to ensure proper structure
+        labelsJson = JSON.stringify(JSON.parse(labelsStr), null, 2);
+      } catch (error) {
+        console.error("Error formatting labels:", error);
+        labelsJson = "{}"; // Default to empty object if error
+      }
+    }
+
+    // Set it in the editor
+    setLabelsJson(labelsJson);
+    setLabelUpdateError("");
+    setIsEditLabelsModalVisible(true);
+  };
+
+  const handleUpdateLabels = async () => {
+    try {
+      // Parse the JSON to validate it
```

Comment:
You should not remove and write a record again. Because we can deal with big records and we don't have enough memory to download it. Please check the https://www.reduct.store/docs/guides/data-management#changing-labels